botocore/client.py

```diff
@@ -606,8 +608,21 @@ def _create_endpoint(
             resolved, region_name, endpoint_url
         )
         if endpoint_url is None:
-            # Use the sslCommonName over the hostname for Python 2.6 compat.
-            hostname = resolved.get('sslCommonName', resolved.get('hostname'))
+            sslCommonName = resolved.get('sslCommonName')
+            hostname = resolved.get('hostname')
+            is_disabled = ensure_boolean(
+                os.environ.get('BOTO_EXPERIMENTAL__DISABLE_COMMONNAME', '')
+            )
+            if not is_disabled and sslCommonName is not None:
+                warnings.warn(
+                    f'The {service_name} client is currently using a '
+                    f'deprecated endpoint: {sslCommonName}. In the next '
+                    f'minor version this will be moved to {hostname}. '
+                    'See https://github.com/boto/botocore/issues/2376 '
```

Comment:
Perhaps it might be better if we create an issue that's a bit more explicit on the what/why surrounding this deprecation. It feels a little weird to link to a closed issue that doesn't really explain the deprecation beyond describing one of the affected services.

Reply:
Yeah that makes sense I'll make one

Reply:
https://github.com/boto/botocore/issues/2705